src/univariates.jl

```diff
@@ -646,7 +646,7 @@ macro _delegate_statsfuns(D, fpre, psyms...)
     pargs = [Expr(:(.), :d, Expr(:quote, s)) for s in psyms]
 
     # output type of `quantile` etc.
-    T = :($D isa DiscreteUnivariateDistribution ? Int : Real)
+    T = :($D <: DiscreteUnivariateDistribution ? Int : Real)
```

Comment:
Is this overwritten somewhere for discrete distributions which are not supported on the integers?

Reply:
No, but the macro that forwards calculations to StatsFuns is only used for discrete distributions with integer valued support.

Reply:
This PR only fixes the issue that was introduced in 0.25.7 when the `eval` parts of the macro where removed but does not introduce any new behaviour.